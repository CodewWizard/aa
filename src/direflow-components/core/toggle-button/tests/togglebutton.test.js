import React from "react";
import renderer from "react-test-renderer";
import ToggleButtonComponent from "../togglebutton";

describe("ToggleButtonComponent", () => {
  const options = [
    { value: "yes", label: "Yes" },
    { value: "no", label: "No" },
  ];

  const props = {
    id: "toggle-btn",
    value: "yes",
    options: options,
    onChange: jest.fn(),
    disabled: false,
    required: true,
  };

  let component;
  beforeEach(() => {
    component = renderer.create(<ToggleButtonComponent {...props} />);
  });

  it("renders correctly", () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
