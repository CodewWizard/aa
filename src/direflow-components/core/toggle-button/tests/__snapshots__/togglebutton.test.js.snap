// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButtonComponent renders correctly 1`] = `
<div
  className="custToggle"
>
  <div
    className="btn-group"
    name="btnGrp_toggle-btn"
    role="group"
    value="yes"
  >
    <label
      className="btn-group-label btn active btn-buttYes"
      disabled={false}
      id="radio-toggle-btn"
      onClick={[Function]}
      required={true}
    >
      <input
        autoComplete="off"
        checked={true}
        disabled={false}
        name="btn_toggle-btn0"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="radio"
        value="yes"
      />
      Yes
    </label>
    <label
      className="btn-group-label btn btn-buttNo"
      disabled={false}
      id="radio-toggle-btn"
      onClick={[Function]}
      required={true}
    >
      <input
        autoComplete="off"
        checked={false}
        disabled={false}
        name="btn_toggle-btn1"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="radio"
        value="no"
      />
      No
    </label>
  </div>
  <input
    id="toggle-btn"
    onInvalid={[Function]}
    required={true}
    style={
      Object {
        "display": "none",
      }
    }
    type="text"
    value="yes"
  />
</div>
`;
